index.py

```diff
@@ -1,80 +1,73 @@
-from flask import Flask, request
-
+from flask import Flask, request, g, json, abort
+from flask_cors import CORS
 import sentry_sdk
 from sentry_sdk.integrations.flask import FlaskIntegration
-    
+from sentry_sdk import configure_scope, capture_exception
+import pdb # set_trace()
+import os
+
+VERSION = os.environ.get("VERSION")
+
 sentry_sdk.init(
-    dsn="https://18562a9e8e3943088b1ca3cedf222e21@sentry.io/1435220",
+    dsn="https://2ba68720d38e42079b243c9c5774e05c@sentry.io/1316515",
+    release=VERSION,
     integrations=[FlaskIntegration()]
 )
-
 app = Flask(__name__)
-
-# request.form .args .files .method
-
-# 1 POST request to /checkout
-# 2 variablize the request's body
-# 3 variablize the request's header X-Transaction-Id
-# 4 review demo spec in notion
-# 5 working endpoint, return response
-# 6 introduce broken code to trigger...
-# 7 graceful error handling
-# 6 return response
+CORS(app)
 
 Inventory = {
-    'wrench': 0,
-    'nails': 0,
+    'wrench': 1,
+    'nails': 1,
     'hammer': 1
 }
 
-# {cart: [{ id: 'nails' }, ...], email: '...@yahoo.com'}
-# param = request.args.get('name')
+
 @app.route('/checkout', methods=['POST'])
 def checkout():
     
-    # POST BODY
-    req_data = request.get_json()
-    
-    # MODULARIZE THIS...
-    email = req_data['email']
-    
-    # CHECKOUT
-    cart = req_data['cart']
+    body = json.loads(request.data)
+
+    # Event Context - make into Flask middleware
+    email = body["email"]
+    transactionId = request.headers.get('X-Transaction-ID')
+    sessionId = request.headers.get('X-Session-ID')
+    with configure_scope() as scope:
+        scope.user = { "email" : email }
+        scope.set_tag("transaction-id", transactionId)
+        scope.set_tag("session-id", sessionId)
+
+    # Checkout
```

Comment:
make its own function. good because adds func to the Stacktrace on the Sentry Event